#!/usr/bin/env python
PACKAGE = "bitbots_vision_tools"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("debug_printer_classes", str_t, 0, "debug_printer_classes", None)

gen.add("pixel_list_path",    str_t,    0, "packagePath",  "/config/testfield_color01.yaml", None)

gen.add("lower_values_h", int_t, 0, "lower_values_h", min=0, max=180)
gen.add("lower_values_s", int_t, 0, "lower_values_s", min=0, max=255)
gen.add("lower_values_v", int_t, 0, "lower_values_v", min=0, max=255)
gen.add("upper_values_h", int_t, 0, "upper_values_h", min=0, max=180)
gen.add("upper_values_s", int_t, 0, "upper_values_s", min=0, max=255)
gen.add("upper_values_v", int_t, 0, "upper_values_v", min=0, max=255)

gen.add("img_msg_topic", str_t, 0, "img_msg_topic", None)
gen.add("img_queue_size", int_t, 0, "img_queue_size", min=0, max=20)

gen.add("color_tester_mask_image", str_t, 0, "color_tester_mask_image", None)

exit(gen.generate(PACKAGE, "color_test", "ColorTest"))
